flocker/testtools/__init__.py

```diff
@@ -215,12 +215,16 @@ def loop(result):
     return d
 
 
-def random_name():
-    """Return a short, random name.
+def random_name(case):
+    """
+    Return a short, random name.
```

Comment:
Docker seems pretty permissive.  I created a container named "x" * 2048 and it was okay.